### PR TITLE
Change the base image in Dockerfile block (nginx:1.17 to nginx:1.24-alpine)

### PR DESCRIPTION
**Pull request by sinsukehlab**
### Why:

nginx:1.17 is vulnerable.

### What's being changed:

README.md
- comment: CC-BY 4.0 &rarr; MIT
- Dockerfile block: nginx:1.17 &rarr; nginx:1.24-alpine
- Add `-p 8080:80` to the `docker run` command

2-add-a-dockerfile.yml
- Verify Dockerfile contents: nginx:1.17 &rarr; nginx:1.24-alpine

### Check off the following:

- [x] For workflow changes, I have verified the Actions workflows function as expected.
- [x] For content changes, I have reviewed the [style guide](https://github.com/github/docs/blob/main/contributing/content-style-guide.md).

### CODE REVIEW

**Review by hectorsector — CHANGES_REQUESTED** [review]
Thanks for making the update @sinsukehlab! We'll also need to update the workflow for step 2 (see below) as it validates the contents of `Dockerfile`. 

To test your changes after you do this, you can:
1. Create a new repository
2. Push from your branch to the new repository's `main` branch
3. Run through the course in the new repository

These steps should flag any remaining problems with the changes. 

https://github.com/skills/publish-packages/blob/530a46efc646ae7d6e0739953b6cb93055143723/.github/workflows/2-add-a-dockerfile.yml#L61-L66

**Comment by sinsukehlab** [comment]
The workflow verified
https://github.com/sinsukehlab/publish-packages-sinsukehlab/actions/runs/4817443439

**Comment by cmwilson21** [comment]
@sinsukehlab Thanks for the verification! I'll take a look at this from the content side ✨

File: README.md [hunk 1]
```diff
@@ -185,14 +186,14 @@ Before we can use this Docker image, you will need to generate a [personal acces
...
+    ```bash
+    docker login ghcr.io -u USERNAME
     ```
-    docker login ghcr.io -u USERNAME -p TOKEN
-    ```
-1. Replace `USERNAME` with your GitHub username
-1. Replace `TOKEN` with the Personal Access Token you just created
-1. Press **Enter**
+1. Replace `USERNAME` with your GitHub username.
+1. Enter the Personal Access Token you just created as the Password.
```
Comment (cmwilson21) [inline README.md:195]:
```suggestion
1. Enter your new Personal Access Token as the password.
```
I think this line could be trimmed down just a little.

File: README.md [hunk 2]
```diff
@@ -229,14 +230,15 @@ Let's trying running it.
...
-   ```
-   docker run -d --rm <YOUR_IMAGE_NAME:TAG>
+   ```bash
+   docker run -dp 8080:80 --rm <YOUR_IMAGE_NAME:TAG>
    ```
 1. Replace `YOUR_IMAGE_NAME` with your image name under the `REPOSITORY` column.
-1. Replace `TAG` with the image tag under the `TAG` column
-   ![example of running the docker command listed above](https://user-images.githubusercontent.com/3250463/219255534-f11b20e8-65de-4f4a-a033-f312ddf507fb.png)
+1. Replace `TAG` with the image tag under the `TAG` column.
+   ![example of running the docker command listed above](https://user-images.githubusercontent.com/3250463/219255534-f11b20e8-65de-4f4a-a033-f312ddf507fb.png)<!-- This screenshot should be changed. -->
```
Comment (cmwilson21) [inline README.md:238]:
```suggestion

```
Could we just remove the image?

File: README.md [hunk 3]
```diff
@@ -229,14 +230,15 @@ Let's trying running it.
...
 1. Press **Enter**.
 1. If everything went well, you will see hash value as output on your screen.
+1. (optional) You can open [localhost:8080](http://localhost:8080) to see the page you just created.
```
Comment (cmwilson21) [inline README.md:241]:
```suggestion
1. Optionally, you can open [localhost:8080](http://localhost:8080) to see the page you just created.
```

**Comment by cmwilson21** [comment]
@sinsukehlab 👋 - Just added a few minor content edits 👍 

Let me know if you accept them and I'll get this merged ✅

**Comment by sinsukehlab** [comment]
@cmwilson21 Thanks for adding the edits. I have accepted them.

**Review by cmwilson21 — APPROVED** [review]
LGTM! Thanks for the updates! 🎉

**Comment by sinsukehlab** [comment]
@cmwilson21 @hectorsector gentle reminder

**Review by hectorsector — APPROVED** [review]
Thanks for your patience @sinsukehlab!

**Pull request merged**